cycode/cli/files_collector/sca/maven/base_restore_maven_dependencies.py

```diff
@@ -39,6 +39,22 @@ def get_manifest_file_path(self, document: Document) -> str:
             else document.path
         )
 
+    def try_restore_dependencies(self, document: Document) -> Optional[Document]:
+        manifest_file_path = self.get_manifest_file_path(document)
+        restore_file_path = build_dep_tree_path(document.path, self.get_lock_file_name())
+
+        if self._verify_restore_file_already_exist(restore_file_path):
+            with open(restore_file_path) as file:
+                restore_file_content = file.read()
+        else:
+            restore_file_content = execute_command(self.get_command(manifest_file_path), manifest_file_path,
+                                                   self.command_timeout)
+
+        return Document(restore_file_path, restore_file_content, self.is_git_diff)
+
+    def _verify_restore_file_already_exist(self, restore_file_path: str) -> bool:
+        return False
```

Comment:
imo we should mark this as an abstract method like others instead without any implementation (`pass`)